src/fsharp/service/ServiceParseTreeWalk.fs

```diff
@@ -459,7 +459,7 @@ module SyntaxTraversal =
 
                 | SynExpr.Match (expr=synExpr; clauses=synMatchClauseList) -> 
                     [yield dive synExpr synExpr.Range traverseSynExpr
-                     yield! synMatchClauseList |> List.map (fun x -> dive x x.RangeOfGuardAndRhs (traverseSynMatchClause path))]
+                     yield! synMatchClauseList |> List.map (fun x -> dive x x.Range (traverseSynMatchClause path))]
```

Comment:
I don't know why `RangeOfGuardAndRhs` was used here. `MatchLambda` and `TryWith` traversals use `Range`, so this change seemed appropriate. Without it `VisitPat` is never fired in a match clause pattern, and so we don't get a chance to create the right completion context.